lib/syskit/log/datastore/normalize.rb

```diff
@@ -244,25 +255,59 @@ def default_index_pathname(logfile_path, index_dir:)
                 Pathname.new(path)
             end
 
+            FOLLOWUP_STREAM_TIME_ERROR_FORMAT =
+                "While building %<stream_name>s, found followup stream whose %<mode>s "\
+                "is before the stream that came before it. Previous sample real time = "\
+                "%<previous>s, sample real time = %<current>s"
+
+            def self.format_timestamp(time_us)
+                Time.at(time_us / 1_000_000).strftime("%Y-%m-%d %H:%M:%S:%6N")
+            end
+
             NormalizationState =
                 Struct
                 .new(:out_io_streams, :control_blocks, :followup_stream_time) do
-                    def validate_time_followup(stream_index, data_block_header)
+                    def report_followup_stream_error(
+                        reporter: NullReporter.new, stream_index:, mode:, previous:,
+                        current:
+                    )
+                        output_stream_name = out_io_streams[stream_index].path
+                        msg = format(
+                            FOLLOWUP_STREAM_TIME_ERROR_FORMAT,
+                            stream_name: output_stream_name, mode: mode,
+                            previous: Normalize.format_timestamp(previous),
+                            current: Normalize.format_timestamp(current)
+                        )
+                        reporter.warn msg
+                    end
+
+                    def validate_time_followup(
+                        stream_index, data_block_header, reporter: NullReporter.new
+                    )
                         # Second part of the followup stream validation (see above)
                         last_stream_time = followup_stream_time[stream_index]
-                        return unless last_stream_time
+                        valid_time = true
+                        return valid_time unless last_stream_time
```

Comment:
`return true`. Setting a value to a constant and then returning it right away just makes things hard to follow.